ui/src/app/app.tsx

```diff
@@ -101,7 +102,13 @@ requests.onError.subscribe(async err => {
         // If basehref is the default `/` it will become an empty string.
         const basehref = document.querySelector('head > base').getAttribute('href').replace(/\/$/, '');
         if (isSSO) {
-            window.location.href = `${basehref}/auth/login?return_url=${encodeURIComponent(location.href)}`;
+            const authSettings = await services.authService.settings();
+
+            if (authSettings?.oidcConfig?.enablePKCEAuthentication) {
+                pkceLogin(authSettings.oidcConfig, getPKCERedirectURI().toString());
```

Comment:
Why is a PKCE login at this point needed? I mean in addition to the existing PKCE login at `login.tsx`

Reply:
This is for when the ID token expires and triggers a re-authentication. Previously when the token expired the user would be redirected to the Dex `/auth/login` endpoint regardless of OIDC configuration. This would fail for PKCE applications as Dex does not support PKCE for upstream connectors: https://github.com/dexidp/dex/pull/3188

To resolve the issue of pkce re-authentication once the token expires, I need a second call to pkceLogin here when the 401 error is caused by the ID Token expiring and pkce authentication is enabled for the oidc client.